packages/react-card/src/components/CardHeader/CardHeader.types.ts

```diff
@@ -1,18 +1,18 @@
-import type { ComponentProps, ComponentState, IntrinsicSlotProps } from '@fluentui/react-utilities';
+import type { ComponentProps, ComponentState, Slot } from '@fluentui/react-utilities';
 
 export type CardHeaderSlots = {
-  root: IntrinsicSlotProps<'div'>;
-  image: IntrinsicSlotProps<'div'>;
-  content?: IntrinsicSlotProps<'div'>;
-  header: IntrinsicSlotProps<'span'>;
-  description: IntrinsicSlotProps<'span'>;
-  action?: IntrinsicSlotProps<'div'>;
+  root: Slot<'div'>;
+  image: Slot<'div'>;
+  content?: Slot<'div'>;
+  header: Slot<'span'>;
+  description: Slot<'span'>;
+  action?: Slot<'div'>;
 };
 
 /**
  * CardHeader props
  */
-export type CardHeaderProps = ComponentProps<CardHeaderSlots>;
+export type CardHeaderProps = ComponentProps<Partial<CardHeaderSlots>>;
```

Comment:
Should all slots be always optional? IMO this should stay as is since design doesn't cover not using an image/header/description, unless it's an architectural decision for them always to be optional.

Reply:
I recently added a description of this change to the PR description (quoted below). It's making the slot _prop_ optional, but the slot is still required on `state`.

> * Some components required adding Partial<> when using ComponentProps. E.g.: `ComponentProps<Partial<AccordionHeaderSlots>>`
>   * This was required because ComponentProps effectively had `Partial<Slots>` built-into it before this change (it added `?` to every slot). After this change, it just includes the `Slots` type directly.
>   * While I could have added an explicit Partial<> inside ComponentProps, that was a bad idea anyways, because it made it difficult to have a required slot prop. Instead, each component can add Partial to the slots if it makes sense in that case.


This PR is actually _fixing_ ComponentProps to correct the problem you describe by making it not add Partial<> internally anymore. But since components were authored before that fix, the simplest solution is to add back Partial<> in components that expected it.

Reply:
I wasn't aware we were internally applying Partial 😬 Then we should remove it for this case.
```suggestion
export type CardHeaderProps = ComponentProps<CardHeaderSlots>;
```

Reply:
Removing Partial will make `<CardHeader />` fail to compile if it doesn't have `image`, `header`, and `description` set -- is that what you intended? In most components, that wasn't the case, which is why I added Partial<> to keep the existing behavior.

Since removing Partial<> would be a change in API for CardHeader, I think I'd prefer if you made that yourself, in a follow-up PR. Is that ok? I don't want to risk this PR being delayed by making too many extra changes.